### PR TITLE
Ignite 16063

### PR DESCRIPTION
**Pull request by alievmirza**

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                        ).yield(),
+                        If.iif(value(partAssignmentsPendingKey).ne(partAssignmentsBytes),
+                                ops(
+                                        put(partAssignmentsPlannedKey, partAssignmentsBytes),
+                                        put(partChangeTriggerKey, ByteUtils.longToBytes(revision))
+                                ).yield(),
+                                ops(remove(partAssignmentsPlannedKey)).yield())),
+                ops().yield());
+
+        metaStorageMgr.invoke(iif);
```
Comment (kgusakov) [inline modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java:89]: This future must be returned as a result for further handling of errors, I believe.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 2]
```diff
@@ -318,10 +318,24 @@ private void onTableCreateInternal(ConfigurationNotificationEvent<TableView> ctx
...
+
+                                        for (int i = 0; i < partCount; i++) {
+                                            String partId = partitionRaftGroupName(
+                                                    ((ExtendedTableConfiguration) tableCfg).id().value(), i);
+
+                                            updateAssignmentsKeys(partId, baselineMgr.baselineNodes(), partCount, newReplicas,
+                                                    replicasCtx.storageRevision(), metaStorageMgr);
+                                        }
 
                                         return CompletableFuture.completedFuture(null);
```
Comment (kgusakov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:336]: I think here we should return anything like [CompletableFuture.allOf](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#allOf-java.util.concurrent.CompletableFuture...-) of all `updateAssignmentsKeys` futures.

File: modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    /**
+     * Key that is needed for the rebalance algorithm.
+     *
+     * @param partId Unique identifier of a partition.
+     * @return Key for a partition.
+     * @see <a href="https://github.com/apache/ignite-3/blob/main/modules/table/tech-notes/rebalance.md">Rebalnce documentation</a>
+     */
+    public static ByteArray partChangeTriggerKey(String partId) {
```
Comment (kgusakov) [inline modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java:99]: `tableId` must be a part of key too, wdyt? Here and in the other keys.

**Review by kgusakov — CHANGES_REQUESTED** [review]

File: modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java (hunk 3 above)
Comment (alievmirza) [reply modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java:99]: In java doc we say that this partId must be unique id of a partition. If you check the usage of this method, you can see that we call `partitionRaftGroupName(UUID tblId, int partition)`, that uses tableId

Comment (kgusakov) [reply modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java:99]: Yep, I see

File: modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java (hunk 1 above)
Comment (alievmirza) [reply modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java]: done

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java (hunk 2 above)
Comment (alievmirza) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java]: done

**Review by kgusakov — APPROVED** [review]

**Pull request merged**